googletest/include/gtest/gtest.h

```diff
@@ -1618,11 +1631,9 @@ AssertionResult CmpHelperOpFailure(const char* expr1, const char* expr2,
 template <typename T1, typename T2>\
 AssertionResult CmpHelper##op_name(const char* expr1, const char* expr2, \
                                    const T1& val1, const T2& val2) {\
-  if (val1 op val2) {\
-    return AssertionSuccess();\
-  } else {\
-    return CmpHelperOpFailure(expr1, expr2, val1, val2, #op);\
-  }\
+  AssertionResult result = (val1 op val2) ? AssertionSuccess() : AssertionFailure();\
+  result << CmpHelperOpMessage(expr1, expr2, val1, val2, #op);\
+  return std::move(result);\
```

Comment:
Do not use `std::move()` on returned object - https://pvs-studio.com/en/docs/warnings/v828/